src/git/git.go

```diff
@@ -104,5 +462,25 @@ func New(log Logger, printElapsedTime ElapsedTime, url, username, password, dest
 
 	gitLabVersioning.repo = repo
 
+	if err := gitLabVersioning.getBranchPointedToHead(); err != nil {
```

Comment:
what do you think put all this check in another function? @esequielvirtuoso

Reply:
I did this before, in a function called "initialize", but I thought you would say... "Why not put this directly in the constructor function"? :D
I'll separate it.